src/metrics/chaoss.ts

```diff
@@ -295,3 +295,66 @@ FORMAT JSONCompact`;
     }
   });
 };
+
+interface IssueResolutionDurationOptions {
+  by: 'open' | 'close';
+  type: 'avg' | 'median';
+  unit: 'week' | 'day' | 'hour' | 'minute';
+}
+export const chaossIssueResolutionDuration = async (config: QueryConfig<IssueResolutionDurationOptions>) => {
+  config = getMergedConfig(config);
+  const whereClauses: string[] = ["type = 'IssuesEvent'"];
+  const repoWhereClause = getRepoWhereClauseForClickhouse(config);
+  if (repoWhereClause) whereClauses.push(repoWhereClause);
+  
+  const endDate = new Date(`${config.endYear}-${config.endMonth}-1`);
+  endDate.setMonth(config.endMonth);  // find next month
+  
+  let by = filterEnumType(config.options?.by, ['open', 'close'], 'open');
+  const byCol = by === 'open' ? 'opened_at' : 'closed_at';
+  let type = filterEnumType(config.options?.type, ['avg', 'median'], 'avg');
+  let unit = filterEnumType(config.options?.unit, ['week', 'day', 'hour', 'minute'], 'day');
+  
+  const sql = `
+SELECT
+  id,
+  argMax(name, time) AS name,
+  ${getGroupArrayInsertAtClauseForClickhouse(config, { key: `resolution_duration`, defaultValue: 'NaN' })}
+FROM
+(
+  SELECT
+    ${getGroupTimeAndIdClauseForClickhouse(config, 'repo', byCol)},
+    round(${type}(dateDiff('${unit}', opened_at, closed_at)), ${config.percision}) AS resolution_duration
+  FROM
+  (
+    SELECT
+      repo_id,
+      argMax(repo_name, created_at) AS repo_name,
+      org_id,
+      argMax(org_login, created_at) AS org_login,
+      issue_number,
+      argMaxIf(action, created_at, action IN ('opened', 'closed' , 'reopened')) AS last_action,
+      maxIf(created_at, action = 'opened') AS opened_at,
+      maxIf(created_at, action = 'closed') AS closed_at
+    FROM github_log.events
+    WHERE ${whereClauses.join(' AND ')}
+    GROUP BY repo_id, org_id, issue_number
+    HAVING ${byCol} >= toDate('${config.startYear}-${config.startMonth}-1') AND ${byCol} < toDate('${endDate.getFullYear()}-${endDate.getMonth() + 1}-1') AND last_action='closed'
```

Comment:
- Filter time range here to avoid losing data for `opened_at` or `closed_at` timestamp in the original query.
- Check `last_action='closed'` to make sure you find a closed issue rather than a reopened issue.